core/src/impl/Kokkos_SharedAlloc.hpp

```diff
@@ -393,6 +393,28 @@ union SharedAllocationTracker {
     return *this;
   }
 
+  /** \brief  Copy assignment without the carry bits logic
+   *         This assumes that externally defined tracking is explicitly enabled
+   */
+  KOKKOS_FORCEINLINE_FUNCTION
+  void assign_direct(const SharedAllocationTracker& rhs) {
+    KOKKOS_IMPL_SHARED_ALLOCATION_TRACKER_DECREMENT
+    m_record_bits = rhs.m_record_bits;
+    KOKKOS_IMPL_SHARED_ALLOCATION_TRACKER_INCREMENT
+  }
+
+  /** \brief  Copy assignment without the initial decrement
+   *         This assumes that current m_record_bits is DO_NOT_DEREF_FLAG
+   *         and externally defined tracking is explicitly disabled */
+  KOKKOS_FORCEINLINE_FUNCTION
+  void assign_force_disable(const SharedAllocationTracker& rhs) {
+    KOKKOS_IMPL_SHARED_ALLOCATION_TRACKER_DECREMENT
```

Comment:
Why are we doing a decrement if tracking is disabled? Also this seems like it might be leading to some unnecessary codegen if it's called in the case where the memory traits are statically known to be unmanaged...

Reply:
the DECREMENT is for the case where a view (or record) is re-assigned to something unmanaged when it was previously managed.  The comment is actually wrong...I fixed it

Reply:
also added a comment above to clarify the assign_direct and assign_force_disabled